types/base45/index.d.ts

```diff
@@ -0,0 +1,20 @@
+// Type definitions for base45 2.0
+// Project: https://github.com/baz/foo (Does not have to be to GitHub, but prefer linking to a source code repository rather than to a project website.)
+// Definitions by: Kyle Hensel <https://github.com/k-yle>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+
+/**
+ * This function takes a base45-encoded string, decodes it, and returns a buffer.
+ */
+export function decode(base45: string): Buffer;
```

Comment:
While the readme says `decode` takes a string, it calls [`Array.from`](https://github.com/irony/base45/blob/main/lib/base45.js#L24) on the input, which means you should be able to also pass at least `Buffer` & `Uint8Array`s in.

Do you think it could be useful to allow those types, given those types are common when doing security and encoding stuff? (I'm not too fussed personally 🙂)

Reply:
I'm not sure what the best option is - since `Buffer`/`Uint8Array` support is undocumented and there are [no tests for it](https://github.com/irony/base45/blob/main/__tests__/encodeDecode.js),  might be better not to allow `Buffer`s or `Uint8Array`s in the typedefs.

Reply:
Yeah, I think probably fine to leave it for now since as you said it's not being tested for so could change in the future - if someone comes along wanting to pass those types in, we can direct them to this + say it's fine but also ideally open an issue on the projects GH asking to add tests to ensure it works in future